app/src/main/java/com/nextcloud/talk/adapters/messages/Reaction.kt

```diff
@@ -39,28 +42,49 @@ class Reaction {
         message: ChatMessage,
         binding: ReactionsInsideMessageBinding,
         context: Context,
-        useLightColorForText: Boolean
+        isOutgoingMessage: Boolean
     ) {
         binding.reactionsEmojiWrapper.removeAllViews()
         if (message.reactions != null && message.reactions.isNotEmpty()) {
 
             var remainingEmojisToDisplay = MAX_EMOJIS_TO_DISPLAY
             val showInfoAboutMoreEmojis = message.reactions.size > MAX_EMOJIS_TO_DISPLAY
             for ((emoji, amount) in message.reactions) {
+                val emojiWithAmountWrapper = LinearLayout(context)
+                emojiWithAmountWrapper.orientation = LinearLayout.HORIZONTAL
+
                 val reactionEmoji = EmojiTextView(context)
                 reactionEmoji.text = emoji
-                binding.reactionsEmojiWrapper.addView(reactionEmoji)
 
-                val reactionAmount = TextView(context)
+                if (message.reactionsSelf != null
+                    && message.reactionsSelf.isNotEmpty()
+                    && message.reactionsSelf.contains(emoji)
+                ) {
+                    emojiWithAmountWrapper.background =
+                        AppCompatResources.getDrawable(context, R.drawable.reaction_self_background)
+                }
+
+                emojiWithAmountWrapper.addView(reactionEmoji)
 
                 if (amount > 1) {
-                    if (useLightColorForText) {
-                        reactionAmount.setTextColor(ContextCompat.getColor(context, R.color.nc_grey))
+                    val reactionAmount = TextView(context)
+                    if (isOutgoingMessage) {
+                        reactionAmount.setTextColor(ContextCompat.getColor(context, R.color.white))
+                    } else {
+                        // reactionAmount.setTextColor(ContextCompat.getColor(context, R.color.nc_message_incoming_reaction_text_color))
+                        reactionAmount.setTextColor(
```

Comment:
atm i fail to get the different color for day/night theme here. any idea what i'm doing wrong @AndyScherzinger ? tried different context etc but i always get black as textcolor....

Reply:
solved, see https://github.com/nextcloud/talk-android/pull/1934/commits/6c1e7c4117569349e3c1f8d293551968fb601a37 - the context from the (parent) view is the one that will be day/night aware